hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUpdateNftsHandlerTest.java

```diff
@@ -219,73 +246,116 @@ void doesntFailWhenMetadataIsEmpty() {
     }
 
     @Test
-    void failsWhenNotSignedByMetadataKey() {
-        final List<Long> serialNumbers = new ArrayList<>(Arrays.asList(1L, 2L));
+    void preHandle_WhenTokenIsNull() {
+        when(preHandleContext.body()).thenReturn(transactionBody);
+        when(transactionBody.tokenUpdateNftsOrThrow()).thenReturn(tokenUpdateNftsTransactionBody);
+        when(tokenUpdateNftsTransactionBody.tokenOrElse(TokenID.DEFAULT)).thenReturn(TokenID.DEFAULT);
+        when(preHandleContext.createStore(ReadableTokenStore.class)).thenReturn(readableTokenStore);
+        when(readableTokenStore.get(TokenID.DEFAULT)).thenReturn(null);
+
+        Assertions.assertThatThrownBy(() -> subject.preHandle(preHandleContext))
+                .isInstanceOf(PreCheckException.class)
+                .has(responseCode(INVALID_TOKEN_ID));
+    }
 
-        final var totalFungibleSupply = 5;
-        writableTokenStore = newWritableStoreWithTokens(Token.newBuilder()
-                .tokenId(TOKEN_123)
-                .tokenType(TokenType.FUNGIBLE_COMMON)
-                .treasuryAccountId(ACCOUNT_1339)
-                .supplyKey((Key) null) // Intentionally missing supply key
-                .totalSupply(totalFungibleSupply)
-                .build());
-        writableTokenRelStore = newWritableStoreWithTokenRels(TokenRelation.newBuilder()
-                .accountId(ACCOUNT_1339)
-                .tokenId(TOKEN_123)
-                .balance(totalFungibleSupply)
-                .build());
-        final var txn = new TokenUpdateNftBuilder()
-                .newNftUpdateTransactionBody(
-                        TOKEN_123, Bytes.wrap("test metadata"), serialNumbers.toArray(new Long[0]));
-        final var context = keyMockContext(txn);
+    @Test
+    void preHandle_WhenTokenHasMetadatayKeySerialsOutsideTreasury() {
+        when(preHandleContext.body()).thenReturn(transactionBody);
+        when(transactionBody.tokenUpdateNftsOrThrow()).thenReturn(tokenUpdateNftsTransactionBody);
+        when(tokenUpdateNftsTransactionBody.tokenOrElse(TokenID.DEFAULT)).thenReturn(nonFungibleTokenId);
+        when(tokenUpdateNftsTransactionBody.serialNumbers()).thenReturn(List.of(1L));
+        when(preHandleContext.createStore(ReadableTokenStore.class)).thenReturn(readableTokenStore);
+        when(preHandleContext.createStore(ReadableNftStore.class)).thenReturn(readableNftStore);
+        when(readableTokenStore.get(nonFungibleTokenId)).thenReturn(token);
+        when(readableNftStore.get(nonFungibleTokenId, 1L)).thenReturn(nft);
+        when(token.tokenIdOrThrow()).thenReturn(nonFungibleTokenId);
+        when(nft.ownerId()).thenReturn(AccountID.newBuilder().accountNum(1).build());
+        when(token.hasMetadataKey()).thenReturn(true);
+
+        assertThatCode(() -> subject.preHandle(preHandleContext)).doesNotThrowAnyException();
```

Comment:
Is this condition sufficient? I don't see how never throwing an exception by itself verifies the new logic (i.e. the code inside the `ifSerialsInTreasury(...)` 'if' block) wasn't executed

Reply:
Added additional verify checks to check the expected invocation of the key to be required.